content/articles/build-a-portfolio-website-with-vite-and-threejs/index.md

```diff
@@ -0,0 +1,449 @@
+### What Is Threejs
+
+Three.js is a JavaScript framework that allows you to easily build a 3d or 2d graphic on your webpage. With Three.js, we don't have to use  WebGL directly. It's essentially an abstract layer built on top of webGL to make it easier to use. Anyone can use a Web browser to view 3D graphics without needing to download any extra framework. Because Three.js is built on JavaScript, integrating interactivity is a breeze. This makes Three.js ideal for creating 3d games on the web platform. It also offers exceptional features such as effects, scenes, cameras, and many more.
+
+ Three.js has been around for a long time. To understand Three.js we must first have an understanding of what  WebGL is. WebGL is a JavaScript API for generating high-performance interactive 3D and 2D graphics.  WebGL can be used without any plug-ins in any supported web browser. WebGL is an OpenGL version based on OpenGL ES (embedded systems). The main drawback of WebGL is that its application development has extensive and heavy code.
+
+Three.js allows you to write short lines of code. With Three.js, you can create video games and showcase products in a 3d form. You can also import into Three.js whenever you want to create a complex object using a 3d software such as blender. To get started with Three.js, we can visit their website [here](https://threejs.org/) and read their documentation.
+
+The main limitation of using Three.js is the performance. We don’t have the same performance between a native application and a web application with WebGL.
+
+### What Is Vitejs?
+
+ It's a build tool for your frontend development. It's a JavaScript development server that significantly improves the frontend development experience. We can simply say it is a build tool for your development. It takes your code and transforms it into something that your browser understands.
+
+Vite allows you to have a development server running.  It also handles and refreshes your files based on what has changed. This process is done extremely fast to reflect your changes on the browser. Other features include:
+
+- Bare module resolving.
+- Hot module replacement.
+- On-demand compilation.
+- Advance configuration.
+
+### Prerequisites 
+
+To follow along with this guide, you need to have:
+- Basic understanding of the JavaScript programming language.
+- A set of image templates
+
+### Building The Portfolio
+
+First, navigate to the folder where you want your project to live and run the following command:
+
+```bash
+npm init @vitejs/app
+```
+After running these instructions, we will be able to name our project. We'll also choose a framework from a list of alternatives. Select `vanilla:vanilla`.
+
+Navigate to the project folder we just created using the `cd` command and install the Three.js package :
+```bash
+ cd your project folder
+```
+Install Three.js
+```bash
+npm install three   
+```
+
+Let’s start up our application:
+
+```bash
+npm run dev
+```
```

Comment:
Add the `http://localhost:3000/` that the reader is supposed to open on a browser.